### DISCUSSION TITLE
Expected timeline for CoreWCF v1.5.0 official release

### DISCUSSION THREAD

**Discussion by devprj22** [CoreWCF/CoreWCF#1228] [Releases]
Hi,

There is a fix pushed in Core WCF v1.5.0. Currently, it is in pre-release and the Latest package is v1.4.1

I wanted to check by when v1.5.0 will be officially released?

Thanks!

**Comment by eaexo** [comment]
Hi @mconnew 

I would also be curios to know when the official release will be available

**Reply by mconnew** [reply]
What feature do you need to come out of preview? When 1.5 is released, some pieces will still be in preview still.

**Reply by devprj22** [reply]
This is the commit I need  - https://github.com/CoreWCF/CoreWCF/commit/0abfcefbd74de83796cfc3caa495642c96ebaba0

**Reply by eaexo** [reply]
NamedPipes and NetTCP, ConfigurationManager

**Comment by mconnew** [comment]
> This is the commit I need - [0abfcef](https://github.com/CoreWCF/CoreWCF/commit/0abfcefbd74de83796cfc3caa495642c96ebaba0)

@devprj22 that made it into 1.4.1
@eaexo, only NamedPipe isn't in 1.4.1. Have you given the preview build a try to make sure it works for you? The 1.5 preview release has some bug fixes that the 1.4 preview didn't have. Now's the time to validate and test it works for you.

**Reply by eaexo** [reply]
@mconnew 
We did - 1.5.0-preview seems to work for us, in any case the question is when 1.5.0 becomes released :-), resp. what is still open or need to be fixed.

**Reply by mconnew** [reply]
The good news is that 1.4.1 and 1.5.0-preview are identical in every way except the version number. So if you hit a bug in anywhere other than the named pipe communication code, (or something only applicable to NamedPipes) that bug will exist in 1.4.1 and will have the same level of support as 1.4.1.  

There's currently 2 pieces missing to go from preview to GA release. I haven't completed the WSDL support so a named pipe service won't advertise itself in the WSDL. The second item is when you stop your service (e.g. WebApplication.StopAsync/Host.StopAsync), the usual WCF close handshake (we have a 1 byte in each direction handshake to cleanly close a channel) isn't done and the client will experience the connection being aborted. Everything else is expected to function correctly.  

There were a few functional bugs in the 1.4.0-preview release that were reported, all but 1 has been fixed. The one which hasn't been fixed also affects NetTcp, and it's when we receive a message larger than the configured MaxReceivedMessageSize, we fail to send a Fault message to the client and uncleanly abort the connection. This is only an issue if the client is configured with a larger MaxReceivedMessageSize than the server as the client won't send a message larger than it's configured for and will throw locally.  

As for timeline, it's got to be a little while until I can get to those. I have an ask from a team internal to Microsoft for ReliableSessions, and I need to finish that before the final couple of tasks for NetNamedPipe.

I hope this info helps you in making your decision.
